### DISCUSSION TITLE
[Feature request] .IMA/.IMG file support for auto_memdisk

### DISCUSSION THREAD

**Discussion by mikolaj612** [ventoy/Ventoy#2030] [Ideas]
Hello.

So far we've got possibility to use it with .ISO files, however I do have some .IMA or .IMG files that I use widely to update BIOS in some older PC's.

Those files came that way directly from manufacturer and I do not want to alter them by any means.
If I press ctrl+d and load them with MEMDISK mode manually they work great - making MEMDISK automatic like for .ISO files would be great functionality for me.

Cheers.

**Comment by steve6375** [comment]
You can change the name of the file
https://ventoy.net/en/doc_name_identifier.html

Does that work or only for ISO files?

xxxx_VTMEMDISK.IMA

**Reply by mikolaj612** [reply]
Thanks! 
It's working - however it's altering file name and in my case those file names are rather long.
Is it possible to alter position names on the list?

**Reply by ventoy** [reply]
You can also use auto memdisk plugin.
https://www.ventoy.net/en/plugin_automemdisk.html

**Reply by steve6375** [reply]
Not sure what you mean? The file name can be shortened or begin with a different character - e.g. # to list it at start of menu.
but must end in _VTMEMDISK.*

**Reply by steve6375** [reply]
@ventoy 
Document for auto memdisk says: https://www.ventoy.net/en/plugin_automemdisk.html
'With this plugin, you can set memdisk mode for some ISO files and Ventoy will always use Memdisk mode to boot them.'
it says ISO files ???  Should it say any payload file ?

**Reply by mikolaj612** [reply]
> You can also use auto memdisk plugin. https://www.ventoy.net/en/plugin_automemdisk.html

I tried but it doesn't work - my files are .IMA with extension changed to .IMG.
However "_VTMEMDISK" option is working if I alter file names.


> Not sure what you mean? The file name can be shortened or begin with a different character - e.g. # to list it at start of menu. but must end in _VTMEMDISK.*

I know that I can alter name in the way I want - however I want to avoid any misunderstanding why some files are named/listed with _VTMEMDISK.
I'm not doing this bootable USB only for myself ;)

I will try to use menu_alias function to see if it work.

**Reply by steve6375** [reply]
If auto memdisk option does not work then you should report it as a bug and clearly expliain what does not work?
e.g.  entry in ventoy.json ? VentoyPlugson.exe?

**Reply by ventoy** [reply]
> @ventoy
> Document for auto memdisk says: https://www.ventoy.net/en/plugin_automemdisk.html
> 'With this plugin, you can set memdisk mode for some ISO files and Ventoy will always use Memdisk mode to boot them.'
> it says ISO files ??? Should it say any payload file ?

It supports ISO and some img files. I have updated the page.

**Comment by mikolaj612** [comment]
I overcome my issue with name_identifier and menu_alias plugins - so technically I did what I was looking for.
However automemdisk still do not work with my .IMA/.IMG files - I'm not calling this an issue by any means, but supporting additional extensions would be nice in the future.

**Reply by ventoy** [reply]
Try this CI release with automemdisk mode.
https://github.com/ventoy/Ventoy/actions/runs/3616465102
https://www.ventoy.net/en/doc_github_ci.html

**Reply by mikolaj612** [reply]
> Try this CI release with automemdisk mode. https://github.com/ventoy/Ventoy/actions/runs/3616465102 https://www.ventoy.net/en/doc_github_ci.html

Thank you.
@ventoy may I ask one thing? Is it possible to use Hiren's BOOT CD 15.2 with Ventoy?
Some stuff like Windows XP is running, however DOS tools aren't working - I use it widely to repair some older computers due to my company regulations.
Ultimate BOOT CD is working fine as far as I tested.

It would be great to know how to launch HBCD properly :)

**Reply by steve6375** [reply]
re HBCD 15.1 ISO, Did you try _VTMEMDISK ? some of the programs work if you load ISO into memory first.
Which specific parts of HBCD 15.2 do you NEED?

**Reply by mikolaj612** [reply]
> re HBCD 15.1 ISO, Did you try _VTMEMDISK ? some of the programs work if you load ISO into memory first. Which specific parts of HBCD 15.2 do you NEED?

I tried but it seems not work anyway.
Some stuff like Vivard for example is unable to work - even if I use memdisk option it's unable to find HBCD location.
From what I quickly tested only some tools are working like Memtest or mini Windows XP for example.

Same stuff happening with YUMI which is somehow corelated with Ventoy.
I know that HBCD is very outdated and probably not a priority enough to apply proper support, but who knows ;)

**Reply by steve6375** [reply]
HBCD 15.2.iso with Ventoy + Vivard works OK on my Asus eeepc. So it does work on some (old?) devices, It also runs on my Asus Z87 mainboard PC.

It almost works on my Lenovo Ideapad 300 - except that when it loads Vivard it hangs at
```
Warning: The high memory area (HMA) is not available
Additional low memory (below 640K) will be used instead
Creating 64MB Ram Drive as R:
```
then eventually I get bad command and syntax error messages and the R:\> DOS prompt.
This seems to be an issue with the IdeaPad 300 BIOS because it also fails if I make a bootable USB drive from the HBCD ISO and boot it without using an ISO file.
